spot-contracts/contracts/FeePolicy.sol

```diff
@@ -178,30 +190,56 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
-    /// @param p Lower, Upper and Growth sigmoid paramters are fixed point numbers with {DECIMALS} places.
-    function updateRolloverFees(RolloverFeeSigmoidParams calldata p) external onlyOwner {
-        require(p.lower >= -int256(SIGMOID_BOUND), "FeeStrategy: sigmoid lower bound too low");
-        require(p.upper <= int256(SIGMOID_BOUND), "FeeStrategy: sigmoid upper bound too high");
-        require(p.lower <= p.upper, "FeeStrategy: sigmoid asymptotes invalid");
-        perpRolloverFee.lower = p.lower;
-        perpRolloverFee.upper = p.upper;
-        perpRolloverFee.growth = p.growth;
+    /// @notice Updates the vault's deployment fee parameters.
+    /// @param vaultDeploymentFee_ The new deployment fee denominated in the underlying tokens.
+    function updateVaultDeploymentFees(uint256 vaultDeploymentFee_) external onlyOwner {
+        vaultDeploymentFee = vaultDeploymentFee_;
+    }
+
+    /// @notice Updates the swap fee percentage for a given pair of assets.
+    /// @dev Setting fees to 100% or 1.0 will effectively pause swapping.
+    /// @param assetIn The asset swapped in.
+    /// @param assetOut The vault asset swapped out.
+    /// @param feePerc The swap fee percentage.
+    function updateVaultSwapFees(
+        address assetIn,
+        address assetOut,
+        uint256 feePerc
+    ) public onlyOwner {
+        require(feePerc <= ONE, "FeeStrategy: perc too high");
+        bytes32 pairHash = keccak256(abi.encodePacked(assetIn, assetOut));
+        if (feePerc > 0) {
+            vaultSwapFeePerc[pairHash] = feePerc;
+        } else {
+            delete vaultSwapFeePerc[pairHash];
+        }
     }
 
     /// @inheritdoc IFeePolicy
-    function computePerpMintFeePerc() external override returns (uint256) {
-        return isOverSubscribed() ? 0 : perpMintFeePerc;
+    function computePerpMintFeePerc(uint256 perpValueIn) external override returns (uint256) {
+        // The act of minting more perps reduces the subscription ratio,
+        // We thus have to check if the "post"-minting subscription state is healthy and
+        // account for fees accordingly.
+        IFeePolicy.SubscriptionState memory postMintState = computeSubscriptionState(
+            perp.getDepositBond(),
+            perp.getTVL() + perpValueIn,
+            vault.getTVL()
+        );
+
+        // When the vault is under-subscribed there exists an active mint fee
+        return (postMintState.normalizedSubscriptionRatio <= 1) ? perpMintFeePerc : 0;
     }
 
     /// @inheritdoc IFeePolicy
     function computePerpBurnFeePerc() external override returns (uint256) {
-        return isOverSubscribed() ? perpBurnFeePerc : 0;
+        IFeePolicy.SubscriptionState memory s = computeSubscriptionState();
+        // When the vault is over-subscribed there exists an active redemption fee
+        return (s.normalizedSubscriptionRatio > 1) ? perpBurnFeePerc : 0;
```

Comment:
It makes sense to check the post op subscription state in the case of perp minting and vault redemption. As these operations will push the system away from the tranche ratio. On the other hand, we know that perp redemption and vault minting push to the system toward the tranche ratio and thus we do not have to check the state.

We currently do not check the post op state after vault redemption because, we charge a fee regardless of the system state. However eventually, I can see us moving toward state dependent vault redemption fee.

Reply:
Ah right. That makes sense.  We should always check the post-op state for all operations (perp mint and redeem, vault mint and redeem). However since the vault fees are not state dependent (are a fixed percentage), we can skip.